packages/block-editor/src/components/block-list/style.scss

```diff
@@ -1020,3 +993,38 @@
 		margin-bottom: auto;
 	}
 }
+
+.components-popover.block-editor-block-list__block__popover {
+	z-index: 29;
```

Comment:
Is it bad to hardcode this?

Reply:
It can be removed, sorry. It's a failed test. :) See https://github.com/WordPress/gutenberg/pull/18779#issuecomment-563281614.

Reply:
Should we remove this now?

Reply:
I've now moved the z-index to the right file. The z-index is still needed to place the toolbar under the header. Before that didn't work, but now that the toolbar is in the correct DOM position, it works.